stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/common/extensions/Channel.kt

```diff
@@ -26,3 +28,18 @@ public fun Channel.getDisplayName(
             .takeIf { it.isNotEmpty() }
         ?: context.getString(devValue)
 }
+
+/**
+ * Returns channel's last regular or system message if exists.
+ *  Deleted and silent messages, as well as messages from shadow-banned users, are not taken into account.
+ *
+ * @return Last message from the channel or null if it doesn't exist
+ */
+public fun Channel.getLastMessage(): Message? =
+    messages.asSequence()
+        .filter { it.createdAt != null || it.createdLocallyAt != null }
+        .filter { it.deletedAt == null }
+        .filter { !it.silent }
+        .filter { it.user.isCurrentUser() || !it.shadowed }
+        .filter { it.isRegular() || it.isSystem() }
+        .maxByOrNull { it.getCreatedAtOrThrow() }
```

Comment:
Seems scary. What about to use here `getCreatedAtOrNull` instead of `getCreatedAtOrThrow`?

Reply:
This should never happen because we are filtering all channels with `createdAt == null` before